src/reflector/compare.hh

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <type_traits>
+
+#include <clean-core/assert.hh>
+#include <clean-core/bit_cast.hh>
+#include <clean-core/has_operator.hh>
+#include <clean-core/move.hh>
+
+#include <reflector/introspect.hh>
+
+namespace rf
+{
+/// returns lhs == rhs if defined, otherwise performs a memberwise comparison
+template <class T>
+[[nodiscard]] bool is_equal(T const& lhs, T const& rhs) noexcept;
+
+/// lhs != rhs
+template <class T>
+[[nodiscard]] bool is_not_equal(T const& lhs, T const& rhs) noexcept
+{
+    return !is_equal(lhs, rhs);
+}
+
+/// lhs < rhs
+template <class T>
+[[nodiscard]] bool is_less(T const& lhs, T const& rhs) noexcept;
+
+/// lhs >= rhs
+template <class T>
+[[nodiscard]] bool is_greater_equal(T const& lhs, T const& rhs) noexcept
+{
+    return !is_less(lhs, rhs);
+}
+
+/// lhs > rhs
+template <class T>
+[[nodiscard]] bool is_greater(T const& lhs, T const& rhs) noexcept
+{
+    return is_less(rhs, lhs);
+}
+
+/// lhs <= rhs
+template <class T>
+[[nodiscard]] bool is_less_equal(T const& lhs, T const& rhs) noexcept
+{
+    return !is_less(rhs, lhs);
+}
+
+namespace detail
+{
+template <class OuterT, class FunctorT>
```

Comment:
just store size_of OuterT as member (now the compare function will be generated for a lot of types)